extensions/amp-analytics/0.1/config.js

```diff
@@ -63,12 +65,62 @@ export class AnalyticsConfig {
   loadConfig() {
     this.win_ = this.element_.ownerDocument.defaultView;
     this.isSandbox_ = this.element_.hasAttribute('sandbox');
+    // eslint-disable-next-line no-undef
+    const fetchVendorConfigPromise = ANALYTICS_VENDOR_SPLIT
+      ? this.fetchVendorConfig_()
+      : Promise.resolve();
 
-    return this.fetchRemoteConfig_()
+    return Promise.all([this.fetchRemoteConfig_(), fetchVendorConfigPromise])
       .then(this.processConfigs_.bind(this))
       .then(() => this.config_);
   }
 
+  /**
+   * Constructs the URL where the given vendor config is located
+   * @private
+   * @param {string} vendor the vendor name
+   * @return {string} the URL to request the vendor config file from
+   */
+  getVendorUrl_(vendor) {
+    const rtv = getMode().rtvVersion;
+    const baseUrl = getMode().localDev ? `/dist` : `${urls.cdn}/rtv/${rtv}`;
+    const max = getMode().minified ? '' : '.max';
+    // bg has a special canary config
+    const canary = vendor === 'bg' && isCanary(self) ? '.canary' : '';
+    return `${baseUrl}/v0/analytics-vendors/${vendor}${canary}${max}.json`;
+  }
+
+  /**
+   * Returns a promise that resolves when vendor config is ready (or
+   * immediately if no vendor config is specified)
+   * @private
+   * @return {!Promise<undefined>}
+   */
+  fetchVendorConfig_() {
+    const type = this.element_.getAttribute('type');
```

Comment:
nit: I found it cleaner to do this.
```
constructor() {
  this.predefinedConfig = {};
}
fetchVendorConfig_() {
  if (! ANALYTICS_VENDOR_SPLIT) {
    this.predefinedConfig_ = ANALYTICS_CONFIG;
    return Promise.resolve();
  }
  ...
}
```
So the `fetchVendorConfig_` method makes sense for both code path. Don't feel strong about this though, up to your implementation.

Reply:
done :)